models/protocols/SendGridProtocol.cfc

```diff
@@ -94,6 +94,14 @@ component extends="cbmailservices.models.AbstractProtocol" {
 
         body[ "personalizations" ] = [ personalization ];
 
+	if ( structkeyExists( mail, 'mailparams' ) && isArray( mail.mailparams ) ){
+		body[ "attachments" ] =  mail.mailParams.map(function(mailParam){
```

Comment:
A `mailParam` can be anything, not just a file.  Here's the [function signature](https://github.com/coldbox-modules/cbmailservices/blob/527a2667507757dd5af3a92a7769ccbe1407ac6f/models/Mail.cfc#L524-L532).  I think you might need to filter the mail params based on the presence of a `file`.

Reply:
you're right and sorry for the delay in the fix